# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/x86/Lowering-x86.h
# Commit: c5ef1152da2b
# Full Hash: c5ef1152da2bd1de3d98b51daa635ac42ff0a153
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-02 04:10:47
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 6.2: Establish dependency from tls for x86 callWithABI div/mod i64. r=lth
#   
#   x86 has few register so to do div/mod for i64 it call the runtime
#   and clobber almost all gp registers including WasmTlsReg.
#   To be able to call c++ runtime via Builtin thunk we need to set up WasmTlsReg.
# ==============================================================================

diff -r f0f44cfd6a07 -r c5ef1152da2b js/src/jit/x86/Lowering-x86.h
--- a/js/src/jit/x86/Lowering-x86.h	Thu Oct 01 07:52:07 2020 +0000
+++ b/js/src/jit/x86/Lowering-x86.h	Thu Oct 01 07:52:24 2020 +0000
@@ -49,7 +49,9 @@
                         MDefinition* rhs);
 
   void lowerDivI64(MDiv* div);
+  void lowerWasmBuiltinDivI64(MWasmBuiltinDivI64* div);
   void lowerModI64(MMod* mod);
+  void lowerWasmBuiltinModI64(MWasmBuiltinModI64* mod);
   void lowerUDivI64(MDiv* div);
   void lowerUModI64(MMod* mod);
 